tuning_test.py

```diff
@@ -0,0 +1,266 @@
+            desired_horizon=100,
+        ),
+    )
+    def test_tune_epistatic_horizon(self, wt_seq, seed, desired_horizon):
+        untuned_landscape = self._get_landscape(wt_seq=wt_seq, seed=seed)
+
+        tuning_kwargs = tuning.get_tuning_kwargs(
+            untuned_landscape,
+            epistatic_horizon=desired_horizon)
+
+        tuned_landscape = self._get_landscape(wt_seq=wt_seq, seed=seed, **tuning_kwargs)
+
+        actual_horizon = tuning.get_epistatic_horizon(tuned_landscape)
+        # TODO(nthomas) make this exact.
+        self.assertAlmostEqual(desired_horizon, actual_horizon, places=3)
+
+# TODO(nthomas) add test for singles and epi offset to act independently...
+
+    @parameterized.named_parameters(
+        dict(
+            testcase_name='tuning_0',
+            seed=2,
+            wt_seq=[0, 0, 0, 0],
+            desired_stats_dict={'fraction_adaptive_singles': 0.7,
+                                'fraction_reciprocal_adaptive_epistasis': 0.5,
+                                'epistatic_horizon': 10}
+        ),
+        dict(
+            testcase_name='tuning_1',
+            seed=3,
+            wt_seq=[1, 1, 1, 1],
+            desired_stats_dict={'fraction_adaptive_singles': 0.2,
+                                'fraction_reciprocal_adaptive_epistasis': 0.66,
+                                'epistatic_horizon': 20},
+        ),
+    )
+    def test_tuned_stats(self, wt_seq, seed, desired_stats_dict):
+        untuned_landscape = self._get_landscape(wt_seq=wt_seq, seed=seed)
+
+        tuning_kwargs = tuning.get_tuning_kwargs(
+            untuned_landscape,
+            normalize_to_singles=True,
+            fraction_adaptive_singles=desired_stats_dict['fraction_adaptive_singles'],
+            fraction_reciprocal_adaptive_epistasis=desired_stats_dict['fraction_reciprocal_adaptive_epistasis'],
+            epistatic_horizon=desired_stats_dict['epistatic_horizon'])
+
+        tuned_landscape = self._get_landscape(wt_seq=wt_seq, seed=seed, **tuning_kwargs)
+        actual_stats_dict = tuning.get_landscape_stats(tuned_landscape)
+
+        num_singles = len(wt_seq) * (untuned_landscape.vocab_size - 1)
+        # Because we are adjusting quantiles, we can only get to within
+        # 1 / num_singles of the desired proportion
+        max_singles_proportion_error = 1.0 / num_singles
+        self.assertBetween(actual_stats_dict['fraction_adaptive_singles'],
+                           desired_stats_dict['fraction_adaptive_singles'] - max_singles_proportion_error,
+                           desired_stats_dict['fraction_adaptive_singles'] + max_singles_proportion_error)
+
+        untuned_fraction_adaptive_singles = tuning.get_landscape_stats(untuned_landscape)['fraction_adaptive_singles']
+        num_adaptive_singles = num_singles * untuned_fraction_adaptive_singles
+
+        num_adaptive_doubles = comb(num_adaptive_singles, 2)
+        # TODO(nthomas) explain that we assume double collisions for safety
```

Comment:
can you flush this out? It wasn't clear to me why you need this.

Reply:
I fleshed out the comment above and changed the error rate to be exact instead of heuristic.